repair/row_level.cc

```diff
@@ -2429,6 +2431,23 @@ future<> repair_service::init_ms_handlers() {
     ms.register_repair_get_diff_algorithms([] (const rpc::client_info& cinfo) {
         return make_ready_future<std::vector<row_level_diff_detect_algorithm>>(suportted_diff_detect_algorithms());
     });
+    ms.register_repair_update_system_table([this] (const rpc::client_info& cinfo, repair_update_system_table_request req) -> future<repair_update_system_table_response> {
+        auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
+        rlogger.debug("repair[{}]: Got repair_update_system_table_request from node={}, range={}, repair_time={}", req.repair_uuid, from, req.range, req.repair_time);
+        auto& db = this->get_db();
+        co_await db.invoke_on_all([&req] (database& local_db) {
+            auto& table = local_db.find_column_family(req.table_uuid);
+            return table.update_repair_time(req.range, req.repair_time);
+        });
+        sstring cql = format("INSERT INTO system.{} (table_uuid, repair_time, repair_uuid, keyspace_name, table_name, range_start, range_end) VALUES (?, ?, ?, ?, ?, ?, ?)",
+                db::system_keyspace::REPAIR_HISTORY);
+        auto range_start = req.range.start() ? req.range.start()->value() : dht::minimum_token();
+        auto range_end = req.range.end() ? req.range.end()->value() : dht::maximum_token();
```

Comment:
You ignore inclusiveness of the bound here, can req.range hold a range of the form [start, end) ?

Reply:
It is not possible. Repair only work with (start, end] ranges. The source of ranges can either be user provided or vnode token ranges. Both of them are in the form of (start, end]. 

Btw, a range defined by C* is:

```
/**
 * A representation of the range that a node is responsible for on the DHT ring.
 *
 * A Range is responsible for the tokens between (left, right].
 *
 * Used by the partitioner and by map/reduce by-token range scans.
 */
public class Range<T extends RingPosition> extends AbstractBounds<T> implements Comparable<Range<T>>, Serializable
```

I think we are adding unnecessary complexity in our implementation of range.hh, which is sad.

Reply:
> It is not possible. Repair only work with (start, end] ranges. The source of ranges can either be user provided or vnode token ranges. Both of them are in the form of (start, end].

Ok. This rule is not documented on dht::token_range, nor on repair_cf_range_row_level(), which means it can be broken in the future. It should be document it if we are relying on it here. It may also make sense to validate it at run time here to be safe.

Reply:
For sure we have to validate it on runtime.

Reply:
I created https://github.com/scylladb/scylla/issues/9615 to track it.